lib/plugins/boot-script.js

```diff
@@ -0,0 +1,90 @@
+var util = require('util');
+var utils = require('../utils');
+var path = require('path');
+var async = require('async');
+var debug = require('debug')('loopback:boot:script');
+var PluginBase = require('../plugin-base');
+var _ = require('lodash');
+
+module.exports = function(options) {
+  return new Script(options);
+};
+
+function Script(options) {
+  PluginBase.call(this, options, 'bootScripts', null);
+}
+
+util.inherits(Script, PluginBase);
+
+Script.prototype.load = function(context) {
+  var options = this.options;
+  var appRootDir = options.rootDir;
+  // require directories
+  var bootDirs = options.bootDirs || []; // precedence
+  bootDirs = bootDirs.concat(path.join(appRootDir, 'boot'));
+  utils.resolveRelativePaths(bootDirs, appRootDir);
+
+  var bootScripts = options.bootScripts || [];
+  utils.resolveRelativePaths(bootScripts, appRootDir);
+
+  bootDirs.forEach(function(dir) {
+    bootScripts = bootScripts.concat(utils.findScripts(dir));
+    var envdir = dir + '/' + options.env;
+    bootScripts = bootScripts.concat(utils.findScripts(envdir));
+  });
+
+  // de-dedup boot scripts -ERS
+  // https://github.com/strongloop/loopback-boot/issues/64
+  bootScripts = _.uniq(bootScripts);
```

Comment:
I wanted to propose extracting a shared function, but I guess it's not worth the effort in this case.